workflows/crisprseq_screening.nf

```diff
@@ -0,0 +1,214 @@
+include { MULTIQC                     } from '../modules/nf-core/multiqc/main'
+include { MAGECK_COUNT                } from '../modules/nf-core/mageck/count/main'
+include { MAGECK_MLE                  } from '../modules/nf-core/mageck/mle/main'
+include { MAGECK_TEST                 } from '../modules/nf-core/mageck/test/main'
+include { CUSTOM_DUMPSOFTWAREVERSIONS } from '../modules/nf-core/custom/dumpsoftwareversions/main'
+include { CRISPRCLEANR_NORMALIZE      } from '../modules/nf-core/crisprcleanr/normalize/main'
+/*
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    RUN MAIN WORKFLOW
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+*/
+
+// Info required for completion email and summary
+def multiqc_report = []
+
+workflow CRISPRSEQ_SCREENING {
+
+    ch_versions = Channel.empty()
+
+    if(!params.count_table){
+        //
+        // SUBWORKFLOW: Read in samplesheet, validate and stage input files
+        //
+        INPUT_CHECK_SCREENING (
+            ch_input
+        )
+        ch_versions = ch_versions.mix(INPUT_CHECK_SCREENING.out.versions)
+
+        //
+        // MODULE: Run FastQC
+        //
+        FASTQC (
+            INPUT_CHECK_SCREENING.out.reads
+        )
+        ch_versions = ch_versions.mix(FASTQC.out.versions.first())
+
+        INPUT_CHECK_SCREENING.out.reads
+        .map { meta, fastq ->
+            [meta.condition, fastq]
+        }
+        .reduce { a, b ->
+            ["${a[0]},${b[0]}", a[1] + b[1]]
+        }
+        .map { condition, fastqs ->
+            [[id: condition], fastqs]
+        }
+        .set { joined }
+
+        //
+        // MODULE: Run mageck count
+        //
+        MAGECK_COUNT (
+            joined,
+            ch_library
+        )
+
+        ch_versions = ch_versions.mix(MAGECK_COUNT.out.versions.first())
+
+
+        MAGECK_COUNT.out.count.map {
+        it -> it[1]
+        }.set { ch_counts }
+
+    } else {
+        Channel.fromPath(params.count_table)
+        .set { ch_counts }
+    }
+
+
+    if(params.crisprcleanr) {
+        ch_crispr_normalize = Channel.of([id: "count_table_normalize"])
+        CRISPRCLEANR_NORMALIZE(
+            ch_crispr_normalize.concat(ch_counts,ch_crisprcleanr).collect(),
+            params.min_reads,
+            params.min_targeted_genes
+        )
+
+        CRISPRCLEANR_NORMALIZE.out.norm_count_file.map {
+            it -> it[1]
+        }.set { ch_counts }
+    }
+
+    if(params.rra_contrasts) {
+        Channel.fromPath(params.rra_contrasts)
+            .splitCsv(header:true, sep:',' )
+            .set { ch_contrasts }
+        counts = ch_contrasts.combine(ch_counts)
+
+        MAGECK_TEST (
+            counts
+        )
+    }
+
+    if(params.mle_design_matrix) {
+        ch_mle = ch_counts.combine(ch_design)
+        ch_mle.map {
+            it -> [[id: it[1].getBaseName()], it[0], it[1]]
```

Comment:
will this line introduce a bug for the same basename files?

Reply:
No it will just write over them normally